app/controllers/portal.js

```diff
@@ -149,6 +76,98 @@ export default Ember.Controller.extend({
      */
     formValidation (validation) {
       this.set('isFormInvalid', validation.errors.length !== 0)
+    },
+    updateDOM: function () {
+      Ember.$('#username_label').text(this.get('username'))
+      Ember.$('#token_label').text(this.get('token'))
+      Ember.$('.general_info').show()
+      Ember.$('.action_generate').show()
+      if (this.get('token') === '~') {
+        Ember.$('.restricted_info').show()
+        Ember.$('.action_unrestrict').show()
+        Ember.$('.action_restrict').hide()
+      } else {
+        Ember.$('.restricted_info').hide()
+        Ember.$('.action_restrict').show()
+        Ember.$('.action_unrestrict').hide()
+      }
+    },
+    createUser: function (element) {
+      return this.get('store').createRecord('developer', {
+        username: element.label,
+        token: element.value
+      })
+      .save()
+      .then((res) => {
+        console.log('For the user with this username: ' + res.get('username') + ', their testing token is: ' + res.get('token'))
+      })
+      .catch((err) => {
+        const data = this.get('data')
+        const index = data.indexOf(element)
+        if (index !== -1) {
+          data.splice(index, 1)
+        }
+        this.set('data', Ember.A(data))
+        Ember.Logger.debug(err)
+      })
+    },
+    /**
+     * Creates a user if they do not exist already and populates them in the selection of data
+     */
+    confirmHandler: function () {
+      const username = this.get('info').username
+      const token = generateToken(30)
+      const element = {
+        label: username,
+        value: token
+      }
+      this.send('createUser', element)
+      const data = this.get('data').toArray()
+      const index = data.indexOf(element)
+      if (index === -1) {
+        data.unshift(element)
+        this.set('data', Ember.A(data))
+        this.set('selectedIndex', [0])
+        this.send('onChangeHandler', token)
+      } else {
+        console.log('This person with username ' + username + 'already exists')
+      }
+    },
+    generateHandler: function () {
+      const token = generateToken(30)
```

Comment:
Magic numbers like `30` should go at the top of the module as a constant so they are obvious and can be easily changed if we decide they should be changed. e.g.
```
const TOKEN_LENGTH=30
...
const token = generateToken(TOKEN_LENGTH)
```